ocaml/xapi/xapi_pool_helpers.ml

```diff
@@ -63,9 +72,11 @@ let valid_operations ~__context record _ref' =
   let current_stack =
     Db.Pool.get_ha_cluster_stack ~__context ~self:(Helpers.get_pool ~__context)
   in
-  if ha_enabled then
-    set_errors Api_errors.ha_is_enabled [] [`ha_enable]
-  else
+  if ha_enabled then (
+    set_errors Api_errors.ha_is_enabled [] [`ha_enable] ;
+    (* TLS verification is not allowed to run if HA is enabled *)
+    set_errors Api_errors.ha_is_enabled [] [`tls_verification_enable]
+  ) else
     set_errors Api_errors.ha_not_enabled [] [`ha_disable] ;
   (* cluster create cannot run during a rolling pool upgrade *)
   if Helpers.rolling_upgrade_in_progress ~__context then
```

Comment:
It may be prudent to block `Pool.enable_tls_verification` when an RPU is in progress